### PR TITLE
07 code edits

### PR DESCRIPTION
**Pull request by restrellado**
Hey @jrosen48 would you mind reviewing and approving these when ready? I got to line 245 but wanted to keep it at a manageable chunk for you to review. If it looks good, go ahead and approve but *don't merge*. I'll keep on suggesting changes on this branch, request review, and will merge at the end for us.

### CODE REVIEW

**Comment by restrellado** [comment]
Just a note on this last set of suggested changes. I saw that you used `distinct()` so I moved my introduction of `distinct()` from the text analysis chapter to this chapter, since it comes first.

**Comment by restrellado** [comment]
There's a reference to a variable called `Gradebook_Item` right around line 411, but `dat` doesn't have that as a variable. Did it get dropped somewhere?

**Comment by jrosen48** [comment]
ahh dang good catch. that was during the synthesis process. I can add this back in easily. For some select variables for one of the dataset, I didn't simulate the data, as the factors had 100s (I think 1000s in some cases!) of levels and I was getting errors (that said I could raise the limit for factors, but that they didn't recommend it). I think I can add this back in not simulated, especially as it's not a student-level var.

**Comment by restrellado** [comment]
> ahh dang good catch. that was during the synthesis process. I can add this back in easily. For some select variables for one of the dataset, I didn't simulate the data, as the factors had 100s (I think 1000s in some cases!) of levels and I was getting errors (that said I could raise the limit for factors, but that they didn't recommend it). I think I can add this back in not simulated, especially as it's not a student-level var.

Ok! I'll let you file an issue and make that change. Any chance it can wait until I finish the code review for this chapter? Might avoid some conflicts of you do that as a separate PR that gets merged after we merge this one.  I'm nearly there–maybe a couple hundred or more lines to go after you approve this latest batch of commits.

**Comment by jrosen48** [comment]
filed an issue for the above in #232 - will wait until code review is complete!

File: 07-wt-ed-ds-pipeline.Rmd [hunk 1]
```diff
@@ -378,11 +384,11 @@ We have a pretty large data frame! Let's take a quick look.
 dat
 ```
 
-It looks like we have nearly 30,000 observations from 30 variables.
+It looks like we have `r nrow(dat)` observations from 30 variables.
```
Comment (jrosen48) [inline 07-wt-ed-ds-pipeline.Rmd]: savvy :)

File: 07-wt-ed-ds-pipeline.Rmd [hunk 2]
```diff
@@ -412,7 +419,9 @@ But--because we are not carrying out a finer-grained analysis using the
 at the student-level (and not at the level of different gradebook items). We can
 do this using the `distinct()` function. This function takes the name of the
 data frame and the name of the variables used to determine what counts as a
-unique case. Another thing to note about `distinct()` is that it will only
+unique case. 
+Imagine having a bucket of Halloween candy that has 100 pieces of candy. You know that these 100 pieces are really just a bunch of duplicate pieces from a relatively short list of candy brands. `distinct` takes that bucket of 100 pieces and returns a bucket containing only one of each distinct piece.
```
Comment (jrosen48) [inline 07-wt-ed-ds-pipeline.Rmd]: <3 this

File: 11-wt-text-analysis.Rmd [hunk 3]
```diff
@@ -268,7 +268,7 @@ The result is a dataset that has status_ids in one column and a positive word fr
 head(pos_tokens$status_id) 
 ```
 
-That's a lot of `status_id`s, many of which are duplicates. Let's try and make the vector of `status_id`s a little shorter. We can get a vector of `status_id`s by using `distinct`. `distinct` is a function that extracts an element from a vector only once. Imagine having a bucket of Halloween candy that has 100 pieces of candy. You know that these 100 pieces are really just a bunch of duplicate pieces from a relatively short list of candy brands. `distinct` takes that bucket of 100 pieces and returns a dataframe where each element is the name of a candy brand. So we can use `distinct` to get a dataframe of `status_id`s, where each `status_id` only appears once:
```
Comment (jrosen48) [inline 11-wt-text-analysis.Rmd]: thanks for moving this here, great idea

**Comment by jrosen48** [comment]
Thanks so much, Ryan, these all look great

File: 07-wt-ed-ds-pipeline.Rmd (hunk 1 above)
Comment (restrellado) [reply 07-wt-ed-ds-pipeline.Rmd]: Yeah I wasn’t sure if we should look into why the number ended up being higher than 30,000 when I ran it yesterday. Might be something to look into if someone else does a code review also.

**Comment by restrellado** [comment]
> Thanks so much, Ryan, these all look great

Welcome! I’m almost through now. If there are any remaining changes I’ll commit them on this branch for you to approve then we’ll go ahead and merge.

**Comment by restrellado** [comment]
Ok all done and ready for you to review. Might be easier if you review changes to 07-wt-ed-ds-pipeline.Rmd starting from line 619 and work backwards to 424. That’ll get you right up to where you left off on the last review.

One big change I wanted to call your attention to: On line 566 I changed `spread` to `pivot_wider` because we used `pivot_longer` a few lines up. I ran the code both ways and compared the structure and number of NAs in the output and everything looks to be the same.

File: 07-wt-ed-ds-pipeline.Rmd [hunk 4]
```diff
@@ -545,11 +557,14 @@ survey_responses <-
...
-  summarise(# Here's where we compute the mean of the responses
+  # Here's where we compute the mean of the responses
+  summarise(
     # Mean response for each measure
-    mean_response = mean(response, na.rm = TRUE)) %>%
+    mean_response = mean(response, na.rm = TRUE)
+    ) %>%
   filter(!is.na(mean_response)) %>%
-  spread(measure, mean_response)
+  pivot_wider(names_from = measure,
```
Comment (jrosen48) [inline 07-wt-ed-ds-pipeline.Rmd]: thanks for adding this - and also adding the names for the arguments to `pivot_wider()` (helpful to me to learn)

**Comment by jrosen48** [comment]
Thank you @restrellado for such a constructive review that improved the chapter a ton!

**Pull request merged**